includes/class-performance.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Performance management.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+use \WP_Error;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Manages settings for Performance.
+ */
+class Performance {
+	/**
+	 * Whether page content is rendered now.
+	 *
+	 * @var bool
+	 */
+	public static $is_rendering_page_content = false;
+
+	/**
+	 * Index of currently rendered Homepage Post Block.
+	 *
+	 * @var int
+	 */
+	public static $current_homepage_posts_block = 0;
+
+	/**
+	 * Add hooks.
+	 */
+	public static function init() {
+		add_filter( 'newspack_blocks_homepage_posts_block_attributes', [ __CLASS__, 'optimise_homepage_posts_block' ] );
+		add_action( 'newspack_theme_before_page_content', [ __CLASS__, 'mark_page_content_rendering_start' ] );
```

Comment:
NIT: This can be accomplished without having to create another hook, with `the_content` filter checking `is_page()`.

Reply:
Checking for page rendering is one of the goals here – the other is ensuring only HP blocks in the _actual_ page content are handled. There might be HP blocks in overlay prompts, which would show up in `the_content` before the actual page content.